vagrant/Vagrantfile

```diff
@@ -1,32 +1,62 @@
-Vagrant.configure("2") do |config|
-  config.vm.box = "bento/ubuntu-17.10"
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+require './hack/lib/cluster-api.rb'
+require './hack/lib/test-guide.rb'
+
+clusterSpec = VagrantCluster            
+    .fromClusterAPI('./spec/*.yaml')              # change the cluster API to customize the cluster topology and select the kubeadm options of interest
+    .ansibleActions(
+        installEtcd: true,                        # this action is performed only if machines with Etcd role are defined            (see ClusterAPI, MachineSets/Role Etcd)
+        setupExternalCA: true,                    # this action is performed only if external certificate authorithy is requested   (see ClusterAPI, Cluster/ProviderConfig)
+        setupControlplaneVip: true,               # this action is performed only if there are 2 or more machines with Master role  (see ClusterAPI, MachineSets/Role Master)
+        kubeadmInit: false, 
+        applyNetwork: false,
+        kubeadmJoinMaster: false,                 # this action is performed only if there are 2 or more machines with Master role  (see ClusterAPI, MachineSets/Role Master)
+        kubeadmJoin: false                        # this action is performed only if machines with Node role are defined            (see ClusterAPI, MachineSets/Role Node)
+    )
+    #.injectKubeadmFromBazelBuild(prefix: nil)    # uncomment if you want to inject a kubeadm release from bazel build; set the prefix if you want to differentiate the binary name
+    #.injectKubeadmFromDockerBuild(prefix: nil)   # uncomment if you want to inject a kubeadm release from docker build; set the prefix if you want to differentiate the binary name
+    #.injectKubeadmFromLocalBuild(prefix: nil)    # uncomment if you want to inject a kubeadm release from local build; set the prefix if you want to differentiate the binary name
```

Comment:
It needs to be possible to load these builds into `/vagrant` out of band from the `vagrant up`.
A dev shouldn't have to reprovision the vm when they make a bugfix or a one-line logging change.
In the previous workflow, the vm is never really recreated because overwriting the build artifact and running `kubeadm reset` is generally enough to get you back to a point where you can rerun tests.

Can we use `vagrant provision --provision-with <..>` for this?
One thing that's nice about the shell script is that it can easily be run from the kubernetes source dir (outside the vagrant directory).

Reply:
done.
Now `kubeadm-playground create` and `kubeadm-playground deploy` are separated commands. There is also a paragraph describing how to handle iterative dev cycles